contracts/Swap.sol

```diff
@@ -124,9 +116,32 @@ contract Swap is OwnerPausable, ReentrancyGuard {
         uint256 _a,
         uint256 _fee,
         uint256 _adminFee,
-        uint256 _withdrawFee,
-        IAllowlist _allowlist
-    ) public OwnerPausable() ReentrancyGuard() {
+        uint256 _withdrawFee
+    ) public {
+        initialize(
```

Comment:
why is this needed? will contracts ever be deployed by constructor?

Reply:
This is for the sake of compatibility. If we ever decide to launch a pool directly from an EOA in 1 transaction, we have the option to do so.

Reply:
hmmm. it's somewhat conceptually messy that the template contract has to be initialized with a full parameter list. Is there a specific use-case for this in mind?

Reply:
let's remove it for now, can always add stuff back later if/when there is an explicit need